src/compiler/c.rs

```diff
@@ -249,11 +249,71 @@ impl<T: CommandCreatorSync, I: CCompilerImpl> Compiler<T> for CCompiler<I> {
     ) -> CompilerArguments<Box<dyn CompilerHasher<T> + 'static>> {
         match self.compiler.parse_arguments(arguments, cwd) {
             CompilerArguments::Ok(mut args) => {
+                // Handle SCCACHE_EXTRAFILES
                 for (k, v) in env_vars.iter() {
                     if k.as_os_str() == OsStr::new("SCCACHE_EXTRAFILES") {
                         args.extra_hash_files.extend(std::env::split_paths(&v))
                     }
                 }
+
+                // Handle cache invalidation for the ROCm device bitcode libraries. Every HIP
+                // object links in some LLVM bitcode libraries (.bc files), so in some sense
+                // every HIP object compilation has an direct dependency on those bitcode
+                // libraries.
+                //
+                // The bitcode libraries are unlikely to change **except** when a ROCm version
+                // changes, so for correctness we should take these bitcode libraries into
+                // account by adding them to `extra_hash_files`.
+                //
+                // See https://clang.llvm.org/docs/HIPSupport.html for details regarding the
+                // order in which the environment variables and command-line arguments control the
+                // directory to search for bitcode libraries.
+                //
+                // In reality, not every bitcode library in that directory is needed, but that is
+                // too much to handle on our side so we just hash every bitcode library in that
+                // directory.
+                if args.language == Language::Hip {
+                    let extract_rocm_arg = |flag: &str| {
```

Comment:
actually, would it be possible to move all this block into a function?

Reply:
Yes, I'm thinking of something like `fn search_hip_device_libs(args: &ParsedArguments, env_vars: &[(OsString, OsString)]) -> Vec<PathBuf>`. Where shall I put this function, in `src/util.rs`?

Reply:
This file is fine